src/Language/PureScript/DCE/CoreFn.hs

```diff
@@ -169,11 +169,23 @@ runDeadCodeElimination entryPoints modules = uncurry runModuleDeadCodeEliminatio
           onBind (Rec es) = concatMap (traverseExpr . snd) es
 
           onBinder :: Binder Ann -> [Key]
-          onBinder (LiteralBinder _ l) = concatMap onBinder (extractLiteral l)
-          onBinder (ConstructorBinder _ _ c _) =
-            [Ident . runProperName <$> c]
-          onBinder (NamedBinder _ _ b1) = onBinder b1
-          onBinder _ = []
+          onBinder b@(LiteralBinder _ l) =
+            foldl'
+              (++)
+              (onBinder' b)
+              (map onBinder (extractLiteral l))
+          onBinder b@(ConstructorBinder _ _ _ bs) =
+            foldl'
+              (++)
+              (onBinder' b)
+              (map onBinder bs)
+          onBinder b@(NamedBinder _ _ b1) = onBinder' b ++ onBinder b1
+          onBinder b = onBinder' b
+
+          onBinder' :: Binder Ann -> [Key]
```

Comment:
I'd love for this to have a more descriptive name.

Reply:
Yeah me too. Is it possible for you to pull in commits from `MaybeJustJames/zephyr` so I can clean up there?